client/ayon_substancepainter/plugins/publish/collect_textureset_images.py

```diff
@@ -156,6 +172,110 @@ def create_image_instance(self, instance, template, outputs,
         image_instance.data["textureSetName"] = texture_set_name
         image_instance.data["textureStackName"] = stack_name
 
+        # Store color space with the instance
+        # Note: The extractor will assign it to the representation
+        colorspace = outputs[0].get("colorSpace")
+        if colorspace:
+            self.log.debug(f"{image_product_name} colorspace: {colorspace}")
+            image_instance.data["colorspace"] = colorspace
+
+        # Store the instance in the original instance as a member
+        instance.append(image_instance)
+
+    def create_image_instance_by_map_filtering(self, instance, outputs,
```

Comment:
Why does this need a completely isolated function? Seems like a LOT of duplicated code? Why can't de-duplicate and only change what is actually different?

What makes this so massively different?